#3551: pwabuilder manifest documentation (#3640)
pwa-builder/PWABuilder: docs/builder/manifest.md

```json
"name": "WebBoard: A Drawing App"
```

### short_name: `string`

`short_name` functions similarly as the `name` field, except that it will only be used when there is not enough character space to display the applications regular name. 

It is recommended that `short_name` be 12 characters or less in length.

```json
"short_name": "WebBoard"
```

?> `short_name` isn't required by Web Standards, but is a required field for packaging with the PWABuilder service.

### description: `string`

`description` is an optional field that can be used to describe the functionality and purpose of your app.

Just like `short_name`, this data should usually align with any store listings.

```json
"description": "WebBoard is your go to application for quick doodles, notes, or sketches!"
```

### background_color: `string`

`background_color` is an optional field that represents the page color of the window that your application will be opened in. This is the color that your app will default to before any styles are loaded. Once styles are loaded, your application will use the background color defined in your CSS.

```json

> "background_color": "green"

Reviewer: background_color and theme_color need to be hex values on PWABuilder so I think that you should put hex values here. You can put a note similar to short name about the difference there but yeah

```

### theme_color: `string`

`theme_color` is an optional field that changes the default color used by certain OS features. For example, this would change the color of your title bar when the application is installed on Windows.

```json
"theme_color": "purple"
```

## Settings

### start_url: `string`

`start_url` is a required field that specifies that URL that will be launched when a user opens your application. This URL can either be an absolute or relative path.

```json
"start_url": "https://docs.pwabuilder.com"
```

### dir: `string`

`dir` is an optional field that specifies the text direction for your PWA. 

It has three values to choose from:

* `auto` - No set directionality for your app.
* `ltr` - Text will go from left to right.
* `rtl` - Text will go from right to left.

```json
"dir": "ltr"
```

### scope: `string`

`scope` is an optional field that defines which URL are within the navigation scope of your application. If the user navigates outside of your app's scope, the will be navigated to a normal browser window.

`scope` can often just be set to the base URL of your PWA.

```json
"scope": "https://docs.pwabuilder.com"
```

> ### language: `string`

Reviewer: I think this is just "lang" not "language"

> # Customizing Your Web Manifest

Reviewer 2: ```suggestion
# Customizing Your Web App Manifest
```

> Many capabilities of progressive web apps are exposed through your app's web manifest, and you can use the PWABuilder service to edit your manifest and enhance your app.

Reviewer 2: ```suggestion
Many capabilities of Progressive Web Apps are exposed through your app's manifest, and you can use the PWABuilder service to edit your manifest and enhance your app.
```

Reviewer 2: nit - just following naming practice

To edit your manifest via the PWABuilder site:

1. Get to your app's report card page. If you need guidance on this, vist the [Quick Start.](/builder/quick-start)

2. In the `Manifest` subsection, click `Edit Your Manifest`.

> 3. A modal with several tabs will open. Each tab will contain different settings for editing your manifest, along with brief tooltips on what each field means.

Reviewer 2: ```suggestion
3. A modal with several tabs will open. Each tab will contain different settings for editing your manifest, along with brief tooltips on what each member means.
```

Reviewer 2: Same here... nit - manifest fields are called members. Not going to correct them all in the rest of the md file. Please correct as you see fit.

4. Once you've edited your manifest to your liking, you can either download your manifest with the `Download Manifest` button at the bottom of the modal, or copy-paste it from the `Code` tab on the far right.

> If you want more in depth documentation on manifest fields, full descriptions for each field follow below. They are separated into the sections they appear in on the PWABuilder website.

Reviewer 2: ```suggestion
If you want more in depth documentation on manifest members, full descriptions for each member follow below. They are separated into the sections they appear in on the PWABuilder website.
```

## Info

### name: `string`

`name` is a required field that specifies the display name for your application. Anywhere where a name for this application would be displayed, this value will be used. 

> This name should usually align with the store listings associate with your applications.

Reviewer 2: "should" and "usually" seem to be contradictive

```json
"scope": "https://docs.pwabuilder.com"
```

### lang: `string`
`lang` is an optional field that specifies the primary language of your app. The `Language` field expects a proper subtag for each langauge, and a list can be found [here.](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry)

```json
"lang": "en"
```

### orientation: `string`

`orientation` is an optional field that specifies the default display orientation for your application. It can be any of the following values:

* `any`
* `natural`
* `portrait`
* `landscape`
* `potrait-primary`
* `portrait-secondary`
* `landscape-primary`
* `landscape-secondary`

```json
"orientation": "portrait"
```

### display: `string`

`display` is an optional field that specifies the display mode that the website should default to.

`display` can take any of the following values:

* `browser`: The applications opens in a standard browser window.
* `minimal-ui`: The application will open in a minimal browser window that still includes certain UI features, such as navigation.
* `standalone`: The application will open in its own window with no browser UI elements.

> * `fullscreen`: The application will make use of all available display space.

Reviewer 2: Should use the same tense. Choose either 'opens in...' or 'will open in...'

`display` will default to `browser` if not specified.

```json
"display": "standalone"
```

### display_override: `Array`

> `display_override` is similar to the `display` field, but allows the developer to select a fallback order for different display modes.

Reviewer 2: Since the rest of file uses 'you' in place for 'the developer'
```suggestion
`display_override` is similar to the `display` field, but allows you to select a fallback order for different display modes.
```

In addition to the four display values above, `display_override` can also take the value `window-control-overlay`. `Window-control-overlay` is a desktop-only display mode and adds a native-style overlay to the top of your application.

```json
"display_override": [
  "window-control-overlay",
  "standalone",
  "browser"
]
```

## Platform

### iarc_rating_id: `string`

> `iarc_rating_id` is an optional field that allows the developer to specify a suitable age range for their application. A rating ID is obtained by answering a questionnaire about an application, and then providing the associated ID for that application.

Reviewer 2: ```suggestion
`iarc_rating_id` is an optional field that allows you to specify a suitable age range for their application. A rating ID is obtained by answering a questionnaire about an application, and then providing the associated ID for that application.
```

You can read more about IARC [here.](https://www.globalratings.com/how-iarc-works.aspx)

```json
"iarc_rating_id": "e58c174a-81d2-5c3c-32cc-34b8de4a52e9"
```

### related_applications: `Array`

`related_applications` is an optional field that specifies applications that have similar or adjacent functionality to your application. This field allows users and store listings to complement your application with related technology.

> This field is an array of application objects, which can each contain a `platform` (the platform that the application is available on), `url` (the web URL where the app can be found), and `id` (the unique ID that specifies the application on the given platform) value. 

Reviewer 2: ```suggestion
This field is an array of application objects, each of which contains a `platform` (the platform that the application is available on), `url` (the web URL where the app can be found), and `id` (the unique ID that specifies the application on the given platform) value. 
```

```json
"prefer_related_applications": true
```

### shortcuts: `Array`

`shortcuts` is an optional field that specifies a list of key tasks within your application. These shortcuts can be displayed by the operating system to allow a user to launch directly to a specific part of the application.

The `shortcuts` field is an array of `shortcut` objects, which can contain the following fields:

*  `name`: The display name of the shorcut. ***Required Field*** 
*  `url`: The url that the shortcut will open to. ***Required Field***
* `short_name`: The shortened display name for when display space is limited.
* `description`: A string description of the shortcut.
* `icons`: A set of icons used to represent the shorcut. This array must include a 96x96 icon.

```json
"shortcuts": [
  {
    "name": "About",
    "url": "/about"
  },
  {
    "name": "Send Message",
    "url": "/new-message",
    "description": "Open a chat with another user and send a message to them."
  }
]
```

### protocol_handlers: `Array`

> `protocol_handlers` is an optional field that specifies an array of protocols that this application can handle. A protocol handler will contain a `protocol` and `url` member to specify how each valid protocol is handled.

Reviewer 2: ```suggestion
`protocol_handlers` is an optional field that specifies an array of protocols that the application can handle. A protocol handler will contain `protocol` and `url` members to specify how each valid protocol is handled.
```

```json
"protocol_handlers": [
  {
    "protocol": "web+music",
    "url": "/play?track=%s"
  }
]
```

?> The `%s` token will be replaced with the URL starting with that protocol handler.

### categories: `Array`

> `categories` is an optional field that specifies an array of categories that this application belongs to. Though this array isn't limited to specific values, you can find a list of known categories [here.](https://github.com/w3c/manifest/wiki/Categories)

Reviewer 2: ```suggestion
`categories` is an optional field that specifies an array of categories that the application belongs to. Though this array isn't limited to specific values, you can find a list of known categories [here.](https://github.com/w3c/manifest/wiki/Categories)
```
